aws/resources/cloudmap_namespace.go

```diff
@@ -28,11 +28,18 @@ func (cns *CloudMapNamespaces) getAll(c context.Context, configObj config.Config
 			return nil, errors.WithStackTrace(err)
 		}
 
-		// Filter namespaces based on configured rules (name patterns, creation time, etc.)
+		// Filter namespaces based on configured rules (name patterns, creation time, tags, etc.)
 		for _, namespace := range page.Namespaces {
+			// Get all tags for the namespace for filtering purposes
+			tags, err := cns.getAllTags(namespace.Arn)
```

Comment:
Making a note here, for very large environments this may be a problem since it will require an API call per namespace. I think the change is good enough for now, but we should this in mind as a possible future improvement.